td-shim-tools/src/tee_info_hash.rs

```diff
@@ -261,6 +258,15 @@ impl TdInfoStruct {
             panic!("The descriptor is not valid!\n");
         }
 
+        raw_image_file
+            .seek(SeekFrom::Start(
+                metadata_off as u64 + size_of::<TdxMetadataGuid>() as u64,
+            ))
+            .unwrap();
+        let mut metadata_buf = vec![0; descriptor.length as usize];
+        raw_image_file.read_exact(&mut metadata_buf).unwrap();
+        let desc = &metadata_buf[0..];
+
```

Comment:
Continuing from above. Then we do
1. read `TdxMetadataDescriptor` and assign the value to variable `desc`

So, seems like the PR does nothing?

Reply:
Previous code can't support OVMF image, it's metadata size is smaller than `0x1000`.
I get the length from TDVF descriptor structure and set the metadata buffer based on it. Now it can locate TDX metadata for OVMF image correctly.